File: scripts/pdf_to_png.py
import fitz  # PyMuPDF
import os

certs = [
    {
        "pdf": r"public\certifications\IMS_Certification.pdf",
        "out": r"public\certifications\ims_certification.png",
        "label": "IMS Certification"
    },
    {
        "pdf": r"public\certifications\AnalyzeSentimentwithNaturalLanguageAPISkillBadge.pdf",
        "out": r"public\certifications\nlp_skill_badge.png",
        "label": "NLP Skill Badge (Google Cloud)"
    }
]

for cert in certs:
    print(f"Converting: {cert['label']} ...")
    try:
        doc = fitz.open(cert["pdf"])
        page = doc[0]  # First page only
        # Render at 2x scale (144 DPI) for a crisp thumbnail
        mat = fitz.Matrix(2.0, 2.0)
        pix = page.get_pixmap(matrix=mat, alpha=False)
        pix.save(cert["out"])
        print(f"  Saved -> {cert['out']} ({pix.width}x{pix.height}px)")
        doc.close()
    except Exception as e:
        print(f"  ERROR: {e}")

print("\nDone!")

File: scripts/pdf-to-png.mjs
import { pdfToPng } from 'pdf-to-png-converter';
import { writeFileSync } from 'fs';
import { resolve } from 'path';

const certs = [
  {
    pdf: resolve('public/certifications/IMS_Certification.pdf'),
    out: resolve('public/certifications/ims_certification.png'),
    label: 'IMS Certification'
  },
  {
    pdf: resolve('public/certifications/AnalyzeSentimentwithNaturalLanguageAPISkillBadge.pdf'),
    out: resolve('public/certifications/nlp_skill_badge.png'),
    label: 'NLP Skill Badge'
  }
];

for (const cert of certs) {
  console.log(`Converting: ${cert.label} ...`);
  try {
    const pages = await pdfToPng(cert.pdf, {
      disableFontFace: false,
      useSystemFonts: true,
      viewportScale: 2.0,   // 2x scale for crisp thumbnail
      pagesToProcess: [1],  // only first page
      strictPagesToProcess: false,
      verbosityLevel: 0
    });
    if (pages.length > 0) {
      writeFileSync(cert.out, pages[0].content);
      console.log(`  ✅ Saved to: ${cert.out}`);
    } else {
      console.log(`  ❌ No pages returned`);
    }
  } catch (err) {
    console.error(`  ❌ Error: ${err.message}`);
  }
}
